src/o365/spfx/commands/project/project-upgrade.ts

```diff
@@ -488,6 +507,78 @@ ${f.resolution}
     return s.join('').trim();
   }
 
+  private getTourReport(findings: FindingToReport[]): string {
+    const tourFindings: FindingTour = {
+      title: "Upgrade Project",
+      steps: []
+    };
+
+    findings.forEach(f => {
+
+      const lineNumber: number = f.position && f.position.line ? f.position.line : this.getLineToModify(f);
```

Comment:
I've noticed a few issues where we incorrectly highlight the affected line. I appreciate you filling in the gap that we have where we not always return the affected line number and you need to determine it yourself. I'd say, let's leave it as-is for now, and add another issue in the repo so that all rules return the affected line number. Once that is in place, we can simplify this logic and always use the location from the finding.